src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/SystemIPGlobalProperties.cs

```diff
@@ -397,6 +397,20 @@ private UnicastIPAddressInformationCollection EndGetUnicastAddresses(IAsyncResul
             return GetUnicastAddressTable();
         }
 
+        public override UnicastIPAddressInformationCollection GetUnicastAddresses()
+        {
+            // Wait for the Address Table to stabilize
+            using (ManualResetEvent stable = new ManualResetEvent(false))
```

Comment:
I decided to keep the stuff in this class the same in order to not diverge from the desktop implementation (currently it's identical).